src/messages/list_readers_response.rs

```diff
@@ -0,0 +1,87 @@
+use crate::messages::message::{MessagePayload, MessageType};
+use anyhow::{Context, Result};
+
+pub struct ListReadersResponse {
+    pub writers: Vec<([u8; 16], u16)>, // Each writer is a tuple of 16-byte IP and 2-byte port
+}
+
+/// Layout of the ListReadersResponse
+/// | 18 bytes per writer | N writers |
+/// |     IP (16 bytes)    | Port (2 bytes) |
```

Comment:
can we also add 2 bytes for the amount of writers at the start of the message thank